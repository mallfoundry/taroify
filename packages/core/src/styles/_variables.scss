// Color Palette
$black: #000;
$black-rgb: 0, 0, 0;
$white: #fff;

$gray-1: #f7f8fa;
$gray-2: #f2f3f5;
$gray-3: #ebedf0;
$gray-4: #dcdee0;
$gray-5: #c8c9cc;
$gray-6: #969799;
$gray-7: #646566;
$gray-8: #323233;

$red: #ee0a24;
$blue: #1989fa;
$cyan: #2db7f5;
$green: #07c160;
$orange: #ff976a;

$orange-dark: #ed6a0c;
$orange-light: #fffbe8;

// Base Unit
$hd: 2 !default;

// Gradient Colors
$gradient-red: linear-gradient(to right, #ff6034, #ee0a24);
$gradient-orange: linear-gradient(to right, #ffd01e, #ff8917);

// Component Colors
$primary-color: var(--blue, $blue);
$info-color: var(--cyan, $cyan);
$success-color: var(--green, $green);
$warning-color: var(--orange, $orange);
$danger-color: var(--red, $red);
$text-color: var(--gray-8, $gray-8);
$text-color-2: var(--gray-6, $gray-6);
$text-color-3: var(--gray-5, $gray-5);
$text-link-color: #576b95;
$active-color: var(--gray-2, $gray-2);
$active-opacity: 0.7;
$disabled-opacity: 0.5;
$background-color: var(--gray-1, $gray-1);
$background-color-light: var(--white, $white);
$text-link-color: #576b95;

// Padding
$padding-base: 4px * $hd;
$padding-xs: $padding-base * 2;
$padding-sm: $padding-base * 3;
$padding-md: $padding-base * 4;
$padding-lg: $padding-base * 6;
$padding-xl: $padding-base * 8;

// Height
$height-xs: 16px * $hd;
$height-sm: 18px * $hd;
$height-md: 20px * $hd;
$height-lg: 22px * $hd;

$font-weight-bold: 500;

// Font
$font-size-xs: 10px * $hd;
$font-size-sm: 12px * $hd;
$font-size-md: 14px * $hd;
$font-size-lg: 16px * $hd;
// Line Height
$line-height-xs: 14px * $hd;
$line-height-sm: 18px * $hd;
$line-height-md: 20px * $hd;
$line-height-lg: 22px * $hd;

$base-font-family: -apple-system, blinkmacsystemfont, 'Helvetica Neue',
helvetica, segoe ui, arial, roboto, 'PingFang SC', 'miui', 'Hiragino Sans GB',
'Microsoft Yahei', sans-serif;

$price-integer-font-family: avenir-heavy, pingfang sc, helvetica neue, arial, sans-serif;

// Animation
$animation-duration-base: 0.3s;
$animation-duration-fast: 0.2s;
$animation-timing-function-enter: ease-out;
$animation-timing-function-exit: ease-in;

// Border
$border-color: var(--gray-3, $gray-3);
$border-width-base: 1px * $hd;
$border-radius-sm: 2px * $hd;
$border-radius-md: 4px * $hd;
$border-radius-lg: 8px * $hd;
$border-radius-max: 999px;
