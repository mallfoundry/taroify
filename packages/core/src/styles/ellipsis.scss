@import './prefix';
@import './mixins/ellipsis';

.#{$component-prefix}ellipsis {
  @include ellipsis();
}

.#{$component-prefix}ellipsis--l2 {
  @include ellipsis(2);
}

.#{$component-prefix}ellipsis--l3 {
  @include ellipsis(3);
}
