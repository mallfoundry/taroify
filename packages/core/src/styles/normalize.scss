@import "./prefix";
@import "./variables";

[class*="#{$component-prefix}"] {
  -webkit-tap-highlight-color: transparent;
}

a {
  text-decoration: none;
}

input,
button,
textarea {
  font: inherit;
  color: inherit;
}

a,
input,
button,
textarea,
[class*="#{$component-prefix}"] {
  &:focus {
    outline: none;
  }
}

ol,
ul {
  padding: 0;
  margin: 0;
  list-style: none;
}
