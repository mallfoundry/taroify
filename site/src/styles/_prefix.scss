$taroify-prefix: "t" + "a" + "r" + "o" + "i" + "f" + "y" + "-";
$component-prefix: $taroify-prefix;
