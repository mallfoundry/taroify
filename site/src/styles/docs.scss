@import "./prefix";
@import "./variables";

.#{$component-prefix}docs {
  &-container {
    box-sizing: border-box;
    padding-left: 220px;
    overflow: hidden;

    &-with-simulator {
      padding-right: 384px;
    }
  }

  @media (min-width: 1680px) {
    &-container {
      width: 1680px;
      margin: 0 auto;
    }
  }

  &-content {
    padding: 30px;
    overflow: hidden;

    a {
      margin: 0 1px;
      color: $vant-blue;
      text-decoration: none;

      &:hover {
        color: darken($vant-blue, 10%);
      }

      &:active {
        color: darken($vant-blue, 20%);
      }
    }

    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      font-weight: normal;
      line-height: 1.5;
      color: $vant-black;

      &[id] {
        cursor: pointer;
      }
    }

    h1 {
      margin: 0 0 30px;
      font-size: 30px;
      cursor: default;
    }

    h2 {
      margin: 45px 0 20px;
      font-size: 25px;
    }

    h3 {
      margin-bottom: 16px;
      font-size: 18px;
      font-weight: 600;
    }

    h4 {
      margin: 24px 0 12px;
      font-size: 16px;
      font-weight: 600;
    }

    h5 {
      margin: 24px 0 12px;
      font-size: 15px;
      font-weight: 600;
    }

    p {
      margin: 16px 0;
      font-size: 15px;
      line-height: 26px;
      color: $vant-text-color;
    }

    table {
      width: 100%;
      margin-top: 12px;
      font-size: 14px;
      line-height: 1.5;
      color: $vant-text-color;
      border-collapse: collapse;

      th {
        padding: 8px 10px;
        font-weight: 600;
        text-align: left;

        &:first-child {
          padding-left: 0;
        }

        &:last-child {
          padding-right: 0;
        }
      }

      td {
        padding: 8px;
        border-top: 1px solid $vant-code-background-color;

        &:first-child {
          padding-left: 0;
          // version tag
          code {
            padding: 2px 6px;
            margin: 0;
            font-size: 11px;
            font-weight: 600;
            color: $vant-blue;
            background-color: fade($vant-blue, 10%);
            border-radius: 20px;
          }
        }

        &:last-child {
          padding-right: 0;
        }
      }

      em {
        font-family: $vant-code-font-family;
        font-size: 14px;
        font-style: normal;
        color: $vant-green;
      }
    }

    ul, ol {
      padding: 0;
      list-style: none;
    }

    ul li,
    ol li {
      position: relative;
      padding-left: 15px;
      margin: 5px 0 5px 10px;
      font-size: 15px;
      line-height: 26px;
      color: $vant-text-color;

      &::before {
        position: absolute;
        top: 0;
        left: 0;
        box-sizing: border-box;
        width: 6px;
        height: 6px;
        margin-top: 10px;
        content: '';
        border: 1px solid $vant-dark-grey;
        border-radius: 50%;
      }
    }

    hr {
      margin: 30px 0;
      border: 0 none;
      border-top: 1px solid #eee;
    }

    p > code,
    li > code,
    table code {
      display: inline;
      padding: 2px 5px;
      margin: 0 2px;
      font-family: inherit;
      font-size: 14px;
      word-break: keep-all;
      background-color: $vant-background-color;
      border-radius: 4px;
      -webkit-font-smoothing: antialiased;
    }

    p > code {
      font-size: 14px;
    }

    section {
      padding: 24px;
      overflow: hidden;
    }

    blockquote {
      padding: 16px;
      margin: 16px 0 0;
      background-color: #ecf9ff;
      border-radius: $vant-border-radius;

      p {
        margin: 0;
      }
    }

    img {
      width: 100%;
      margin: 16px 0;
      border-radius: $vant-border-radius;
    }
  }

  &-card {
    padding: 24px;
    margin-bottom: 24px;
    background-color: #fff;
    border-radius: $vant-border-radius;
    box-shadow: 0 8px 12px #ebedf0;

    h3 {
      margin-top: 0;
    }
  }
}
