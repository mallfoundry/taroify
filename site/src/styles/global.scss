@import "./variables";

body {
  min-width: 1100px;
  font-family: 'Open Sans', -apple-system, BlinkMacSystemFont, 'Helvetica Neue',
  Helvetica, Segoe UI, Arial, Roboto, 'PingFang SC', 'miui',
  'Hiragino Sans GB', 'Microsoft Yahei', sans-serif;
  background-color: $vant-background-color;
}
