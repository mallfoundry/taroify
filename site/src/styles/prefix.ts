const COMPONENT_PREFIX = "\u0074\u0061\u0072\u006f\u0069\u0066\u0079-"

export function prefixClassname(component: string) {
  return `${COMPONENT_PREFIX}${component}`
}
