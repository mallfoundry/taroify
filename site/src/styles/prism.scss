@import "./variables";

.gatsby-highlight-code-line {
  display: block;
  padding-right: 1em;
  padding-left: 1.25em;
  background-color: hsla(207, 95%, 15%, 1);
  border-left: 0.25em solid $vant-green;
}

.gatsby-highlight {
  position: relative;
  display: block;
  padding: 16px;
  margin: 20px 0 0;
  overflow-x: auto;
  font-family: $vant-code-font-family;
  font-size: 14px;
  font-weight: 400;
  line-height: 26px;
  color: $vant-code-color;
  word-wrap: break-word;
  white-space: pre-wrap;
  background-color: #fafafa;
  border-radius: 16px;
}

code[class*='language-'], pre[class*='language-'] {
  padding: 0;
  margin: 0;
  color: $vant-code-color;
  background: transparent;
}

.token.subst {
  color: $vant-code-color;
}

.token.string,
.token.meta,
.token.symbol,
.token.attr-value,
.token.template-tag,
.token.template-variable,
.token.addition {
  color: $vant-green;
}


.token.punctuation {
  color: $vant-code-color;
}

.token.comment,
.token.quote {
  color: #999;
}

.token.params,
.token.keyword,
.token.property,
.token.attribute {
  color: $vant-purple;
}

.token.function,
.token.deletion,
.token.variable,
.token.boolean,
.token.number,
.token.regexp,
.token.literal,
.token.bullet,
.token.link {
  color: #eb6f6f;
}

.token.class-name,
.token.tag,
.token.operator,
.token.attr-name,
.token.selector,
.token.selector-tag,
.token.title,
.token.section,
.token.built_in,
.token.doctag,
.token.type,
.token.name,
.token.selector-id,
.token.selector-class,
.token.strong {
  color: #4994df;
}

.token.emphasis {
  font-style: italic;
}
