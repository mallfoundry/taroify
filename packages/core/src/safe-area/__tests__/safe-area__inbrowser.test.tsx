import { render } from "@testing-library/react"
import * as React from "react"
import SafeArea from "../safe-area"
import { prefixClassname } from "../../styles"

jest.mock("../../utils/base", () => ({
  inBrowser: true,
}))

describe("<SafeArea /> in browser", () => {
  it("renders in browser without position", () => {
    const { container } = render(<SafeArea>内容</SafeArea>)

    expect(container.querySelector(`.${prefixClassname("safe-area")}`)).not.toBeNull()
    expect(container.querySelector(`.${prefixClassname("safe-area--top")}`)).toBeNull()
    expect(container.querySelector(`.${prefixClassname("safe-area--bottom")}`)).toBeNull()
    const styles = getComputedStyle(container.firstChild as Element)
    expect(Boolean(styles.paddingTop)).toBe(false)
    expect(Boolean(styles.paddingBottom)).toBe(false)
    expect(container).toMatchSnapshot()
  })

  it("renders top in browser", () => {
    const { container } = render(<SafeArea position="top">顶部安全区</SafeArea>)

    expect(container.querySelector(`.${prefixClassname("safe-area--top")}`)).not.toBeNull()
    expect(container.firstChild).toHaveStyle({
      paddingTop: expect.stringMatching(/^(env|constant)\(safe-area-inset-top\)$/),
    })
    const styles = getComputedStyle(container.firstChild as Element)
    expect(Boolean(styles.paddingBottom)).toBe(false)

    expect(container).toMatchSnapshot()
  })

  it("renders bottom in browser", () => {
    const { container } = render(<SafeArea position="bottom">底部安全区</SafeArea>)

    expect(container.querySelector(`.${prefixClassname("safe-area--bottom")}`)).not.toBeNull()
    expect(container.firstChild).toHaveStyle({
      paddingBottom: expect.stringMatching(/^(env|constant)\(safe-area-inset-bottom\)$/),
    })
    const styles = getComputedStyle(container.firstChild as Element)
    expect(Boolean(styles.paddingTop)).toBe(false)

    expect(container).toMatchSnapshot()
  })
})
