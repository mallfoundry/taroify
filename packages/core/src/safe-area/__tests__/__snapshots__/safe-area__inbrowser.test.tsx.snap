// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SafeArea /> in browser renders bottom in browser 1`] = `
<div>
  <taro-view-core
    class="taroify-safe-area taroify-safe-area--bottom"
  >
    底部安全区
  </taro-view-core>
</div>
`;

exports[`<SafeArea /> in browser renders in browser without position 1`] = `
<div>
  <taro-view-core
    class="taroify-safe-area"
  >
    内容
  </taro-view-core>
</div>
`;

exports[`<SafeArea /> in browser renders top in browser 1`] = `
<div>
  <taro-view-core
    class="taroify-safe-area taroify-safe-area--top"
  >
    顶部安全区
  </taro-view-core>
</div>
`;
